### PR TITLE
Adjust dynamic fanout/fanin example

### LINKED ISSUE #743: Example dynamic_fanout_fanin.py doesn't work
<h2>Pre-bug-report checklist</h2>

**1. This bug can be reproduced using YAML**
- [ ] Yes [👉 Please report a bug to the Argo Workflows GitHub 👈](https://github.com/argoproj/argo-workflows/issues/new/choose)
- [x] No

**2. This bug occurs when...**
- [x] running Hera code without submitting to Argo (e.g. when exporting to YAML)
- [x] running Hera code and submitting to Argo

<h2>Bug report</h2>

**Describe the bug**
_A clear and concise description of what the bug is:_
The example examples/workflows/dynamic_fanout_fanin.py doesn't work

_Error log if applicable_:
```
Traceback (most recent call last):
  File "/path/file.py", line 43, in <module>
    w.create()
  File "/path/.venv/lib/python3.10/site-packages/hera/workflows/workflow.py", line 385, in create
    wf = self.workflows_service.create_workflow(
  File "/path/.venv/lib/python3.10/site-packages/hera/workflows/service.py", line 926, in create_workflow
    raise exception_from_server_response(resp)
hera.exceptions.InternalServerError: Server returned status code 500 with message: `templates.d.tasks.fanin failed to resolve {{tasks.fanout.outputs.parameters}}`
```

**To Reproduce**
Full Hera code to reproduce the bug:
```py
from hera.workflows import DAG, Parameter, Workflow, script
from hera.workflows.models import ValueFrom


@script()
def generate():
    import json
    import sys

    json.dump([{"value": i} for i in range(10)], sys.stdout)


@script(outputs=[Parameter(name="value", value_from=ValueFrom(path="/tmp/value"))])
def fanout(object: dict):
    print("Received object: {object}!".format(object=object))
    # Output the content of the "value" key in the object
    value = object["value"]
    with open("/tmp/value", "w") as f:
        f.write(str(value))


@script()
def fanin(values: list):
    print("Received values: {values}!".format(values=values))


# assumes you used `hera.set_global_token` and `hera.set_global_host` so that the workflow can be submitted
with Workflow(generate_name="dynamic-fanout-fanin", entrypoint="d") as w:
    with DAG(name="d"):
        g = generate()
        fout = fanout(with_param=g.result)
        fin = fanin(arguments=fout.get_parameters_as("values"))
        g >> fout >> fin
w.create()
```

**Expected behavior**
_A clear and concise description of what you expected to happen:_
The example should work. Using w.to_file() and looking at the YAML, the error is '{{tasks.fanout.outputs.parameters}}' when it should be '{{tasks.fanout.outputs.parameters.value}}' .

**Environment**
 - OS: Win10
 - Browser: Firefox
 - Version of Hera: present in both 5.6.0 and 5.2.0
 - Version of Argo: 3.2.4

**Additional context**

### PR DESCRIPTION
**Pull request by flaviuvadan**
**Pull Request Checklist**
- [x] Fixes #743
- [ ] Tests added
- [ ] Documentation/examples added
- [x] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

### CODE REVIEW

File: examples/workflows/upstream/suspend-template.upstream.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ spec:
 
   - name: delay
     suspend:
-      duration: "20"    # Must be a string. Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
```
Comment (dejamiko) [inline examples/workflows/upstream/suspend-template.upstream.yaml:35]: Is this change intentional?

**Review by dejamiko — COMMENTED** [review]
Looking good, similarly to the other PR, one question

**Review by samj1912 — APPROVED** [review]
